src/mellowgate/core/discrete.py

```
@@ -155,7 +164,66 @@ class DiscreteProblem:
+
+        # Test the function with sample inputs to ensure it works correctly
+        try:
+            test_probs = jnp.array([0.3, 0.7])
+            test_key = jax.random.PRNGKey(42)
+            result = self.sampling_function(test_probs, test_key)
+
+            # Validate the result
+            if not isinstance(result, jnp.ndarray):
+                raise ValueError(
+                    f"Sampling function must return a jnp.ndarray, got {type(result)}"
+                )
```

Ruff (0.13.2)

Prefer `TypeError` exception for invalid type

(TRY004)